### PR TITLE
[DE-248] DE-399 Fix errors/warning in Stoplight

### PR DESCRIPTION
**Pull request by michalpierog**
This PR fixes all errors and warnings showed in Stoplight:
- fix nullable properties
- fix examples
- reuse page/per_page parameter
- add missing tags

### CODE REVIEW

File: reference/Chargify-API.v1.yaml [hunk 1]
```diff
@@ -2729,79 +2661,72 @@ paths:
...
           in: query
           description: 'The start date and time (format YYYY-MM-DD HH:MM:SS) with which to filter the date_field. Returns coupons with a timestamp at or after exact time provided in query. You can specify timezone in query - otherwise your site''s time zone will be used. If provided, this parameter will be used instead of start_date.'
           name: 'filter[start_datetime]'
         - schema:
             type: array
-            example: 'filter[ids]=1,2,3'
+            example:
+              - 1
+              - 2
+              - 3
```
Comment (maciej-nedza) [inline reference/Chargify-API.v1.yaml:2704]: now it renders as `[1,2,3]`, before it was `1,2,3`. Will adding `[]` work correctly when calling the api?

File: reference/Chargify-API.v1.yaml [hunk 2]
```diff
@@ -2729,79 +2661,72 @@ paths:
...
             minItems: 1
           in: query
           name: 'filter[ids]'
           description: Allows fetching coupons with matching id based on provided values.
+          style: form
```
Comment (maciej-nedza) [inline reference/Chargify-API.v1.yaml:2709]: what is it needed for?

Comment (michalpierog) [reply reference/Chargify-API.v1.yaml:2709]: This describes how complex parameters are serialised:
https://swagger.io/docs/specification/serialization/

File: reference/Chargify-API.v1.yaml (hunk 1 above)
Comment (michalpierog) [reply reference/Chargify-API.v1.yaml]: Fixed - we decided to add description how to use in query so we can keep examples valid.

File: reference/Chargify-API.v1.yaml [hunk 3]
```diff
@@ -3121,58 +3049,76 @@ paths:
...
             minItems: 1
-            example: 'filter[codes]=free,free_trial'
+            example:
+              - free
+              - free_trial
           in: query
           name: 'filter[codes]'
-          description: Allows fetching coupons with matching code based on provided values.
+          description: 'Allows fetching coupons with matching code based on provided values. 
+          Use in query: `filter[ids]=1,2,3`.'
```
Comment (maciej-nedza) [inline reference/Chargify-API.v1.yaml:3072]: I'd make it match example here, `filter[ids]=free,free_trial `

Comment (michalpierog) [reply reference/Chargify-API.v1.yaml:3072]: Fixed

**Review by maciej-nedza — APPROVED** [review]
LGTM

**Pull request merged**